test/common/index.js

```diff
@@ -690,12 +690,26 @@ Object.defineProperty(exports, 'hasIntl', {
 });
 
 // Useful for testing expected internal/error objects
+const kAcceptableErrorTypes = ['NodeError', 'AssertionError', 'RangeError'];
 exports.expectsError = function expectsError({code, type, message}) {
-  return function(error) {
+  const expectedTypeName = type && type.name;
+  return function expectsErrorCheck(error) {
     assert.strictEqual(error.code, code);
-    if (type !== undefined)
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
+    if (expectedTypeName) {
```

Comment:
if I do `common.expectsError({type: TypeError})` it *will* only be `true` if the error is a `TypeError` (or a subclass of `TypeError`. If you'd like a stricter check, perhaps add a new argument like `{strictType: TypeError}` so that tests that might happen to require absolutely strict checking can do so. Or, perhaps even allow `type:` to take a function so stricter checking can be optionally performed.